ethcore/src/client/client.rs

```diff
@@ -1535,6 +1499,52 @@ impl Drop for Client {
 	}
 }
 
+fn transaction_receipt(transaction: Option<LocalizedTransaction>, receipts: Option<Vec<Receipt>>) -> Option<LocalizedReceipt> {
+	let tx_and_sender = transaction
+		.and_then(|tx| tx.sender().ok().map(|sender| (tx, sender)));
```

Comment:
having this check within the body (and corresponding match for the output) is a bit weird.
`fn transaction_receipt(LocalizedTransaction, Address, Vec<Receipt>) -> LocalizedReceipt` seems like a cleaner interface.

a comment explaining what it does would be nice too. although it seems to me that actually you just need a `fn(Transaction, Address, Receipt, usize)` to do the same thing.